Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -16,16 +16,85 @@
 
 #include "Firestore/core/src/firebase/firestore/remote/datastore.h"
 
+#include <unordered_set>
+
 #include "Firestore/core/include/firebase/firestore/firestore_errors.h"
+#include "Firestore/core/src/firebase/firestore/core/database_info.h"
+#include "Firestore/core/src/firebase/firestore/remote/grpc_completion.h"
+#include "Firestore/core/src/firebase/firestore/util/executor_libdispatch.h"
 #include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_cat.h"
 
 namespace firebase {
 namespace firestore {
 namespace remote {
 
-util::Status Datastore::ConvertStatus(grpc::Status from) {
+using core::DatabaseInfo;
+using util::AsyncQueue;
+using util::Status;
+using util::internal::Executor;
+using util::internal::ExecutorLibdispatch;
+
+namespace {
+
+std::unique_ptr<Executor> CreateExecutor() {
+  auto queue = dispatch_queue_create("com.google.firebase.firestore.datastore",
```

Comment:
nit: This name of this class dates from when we didn't know what the backend would be called so we just called it what it was then. The name doesn't matter from the point of view of any customer we have.

Queue names *are* customer visible though, so we should name this something less historical and more accurate.

Only the last segment is problematic. Maybe "rpc" or "network" instead?